src/libs/actions/Policy.ts

```diff
@@ -2503,6 +2507,60 @@ function clearCategoryErrors(policyID: string, categoryName: string) {
     });
 }
 
+function deleteWorkspaceCategories(policyID: string, categoryNamesToDelete: string[]) {
+    const policyCategories = allPolicyCategories?.[`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`] ?? {};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: categoryNamesToDelete.reduce<Record<string, Partial<PolicyCategory>>>((acc, categoryName) => {
+                    acc[categoryName] = {pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE};
+                    return acc;
+                }, {}),
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.SET,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: Object.keys(policyCategories).reduce<PolicyCategories>((acc, key) => {
+                    if (!categoryNamesToDelete.includes(key)) {
+                        acc[key] = policyCategories[key];
+                    }
+
+                    return acc;
+                }, {}),
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: Object.keys(policyCategories).reduce<PolicyCategories>((acc, key) => {
```

Comment:
Same here, we can just iterate over `categoryNamesToDelete` and remove pendingAction and add the error